awx/api/views.py

```diff
@@ -525,14 +560,37 @@ class InstanceList(ListAPIView):
     new_in_320 = True
 
 
-class InstanceDetail(RetrieveAPIView):
+class InstanceDetail(RetrieveUpdateAPIView):
 
     view_name = _("Instance Detail")
     model = Instance
     serializer_class = InstanceSerializer
     new_in_320 = True
 
 
+    def update(self, request, *args, **kwargs):
```

Comment:
I think I might move this logic down to the model since it's pretty close to what we do over in the heartbeat task